test/integration/isolated/e2e.go

```diff
@@ -0,0 +1,164 @@
+//go:build integration_tests
+
+package isolated
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	ktfkongaddon "github.com/kong/kubernetes-testing-framework/pkg/clusters/addons/kong"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+func NameFromT(t *testing.T) string {
+	t.Helper()
+
+	name := strings.ToLower(t.Name())
+	name = strings.ReplaceAll(name, "_", "-")
+	name = strings.ReplaceAll(name, "/", ".")
+
+	return name
+}
+
+// CreateNSForTest creates a random namespace with the runID as a prefix. It is stored in the context
+// so that the deleteNSForTest routine can look it up and delete it.
+func createNSForTest(ctx context.Context, cfg *envconf.Config, t *testing.T, runID string) (context.Context, error) {
+	t.Helper()
+
+	ns := RandomName("ns-"+runID, 10)
+	ctx = context.WithValue(ctx, GetNamespaceKey(t), ns)
+
+	t.Logf("Creating NS %v for test %v", ns, t.Name())
+	nsObj := corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ns,
+			Labels: map[string]string{
+				"kubernetes-ingress-controller.konghq.com/test-name": NameFromT(t),
+				"kubernetes-ingress-controller.konghq.com/run-id":    runID,
+				"kubernetes-ingress-controller.konghq.com/test-type": "integration",
+			},
+		},
+	}
+	c, err := client.New(cfg.Client().RESTConfig(), client.Options{})
+	if err != nil {
+		return ctx, err
+	}
+	return ctx, c.Create(ctx, &nsObj)
+
+	// TODO: cfg.Client().Resources() causes a data race when run in parallel
```

Comment:
Yeah... You got me motivated to create an issue upstream so that we can properly reference it here: https://github.com/kubernetes-sigs/e2e-framework/issues/352

Hope that works :)

Reply:
Yay, perfect! :)